utils_linux.go

```diff
@@ -123,11 +123,13 @@ func setupIO(process *libcontainer.Process, container *libcontainer.Container, c
 			}
 			uc, ok := conn.(*net.UnixConn)
 			if !ok {
+				conn.Close()
```

Comment:
Thank you for your contribution.
﻿
In practice, omitting these close operations doesn't cause any harm, since runc will exit anyway if an error occurs during the setupIO call.
﻿
However, from a code style and best practices perspective, I think we should always close the socket connection once an error is encountered after the connection has been established.
﻿
Considering both aspects, LGTM.